qt/widgets/common/inc/MantidQtWidgets/Common/MuonPeriodInfo.h

```diff
@@ -0,0 +1,66 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+#include "DllOption.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "ui_MuonPeriodInfo.h"
+
+#include <QWidget>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+/**
+ * A widget used in the Muon GUI's to display period information in a readable format
+ */
+class EXPORT_OPT_MANTIDQT_COMMON MuonPeriodInfo : public QWidget {
+  Q_OBJECT
+
+public:
+  /// Reads the data of the sample log from the workspace
+  static std::string readSampleLog(Mantid::API::MatrixWorkspace_sptr ws, const std::string &logName);
+  /// Splits a string separated by a delimeter
+  static std::vector<std::string> parseSampleLog(const std::string &log, const std::string &delim);
+  /// Unifies the length of all vectors given
+  static std::vector<std::vector<std::string>> makeCorrections(std::vector<std::vector<std::string>> &logs);
+  explicit MuonPeriodInfo(QWidget *parent = nullptr);
+  /// Add a period to the table on the widget
+  void addPeriodToTable(const std::string &name, const std::string &type, const std::string &frames,
+                        const std::string &totalFrames, const std::string &counts, const std::string &tag);
+  /// Gets all sample log data related to periods
+  std::vector<std::vector<std::string>> getInfo(Mantid::API::MatrixWorkspace_sptr ws);
+  /// Takes the workspace and adds it's period info to the table if any
+  void addInfo(const Mantid::API::Workspace_sptr ws);
+  /// Set the title of the widget
+  void setWidgetTitleRuns(const std::string &title);
+  /// Get the title of the widget
+  std::string getWidgetTitleRuns() const;
+  /// Set the number of sequences that was gathered
```

Comment:
what do you mean by sequence?

Reply:
To be honest this is just the name given in the nexus file, I am not 100% sure what is meant by it

Reply:
could you ask Steve?

Reply:
From Steve:
_The entry period_sequences records how many times the complete period sequence was executed during the run. For instance, if we look at EMU run 114295, which was a run carried out with a two period sequence collecting 500 frames in each period (i.e. 1000 frames to complete a single sequence), period_sequences reads 218 suggesting 218000 frames have been collected during the run. This compares well with frames_good (218002), a record of the total number of frames collected. (not sure why we’ve recorded two additional frames in the final period, but that’s another matter)._

Reply:
So in simple terms: a sequence is completing recording for all of the periods once?

Can we add a comment explaining this?